provider/aci_test.go

```diff
@@ -1191,3 +1191,151 @@ func TestCreatePodWithProjectedVolume(t *testing.T) {
+	secretLister := NewMockSecretLister(mockCtrl)
+	secretNamespaceLister := NewMockSecretNamespaceLister(mockCtrl)
+	configMapLister := NewMockConfigMapLister(mockCtrl)
+	serviceLister := NewMockServiceLister(mockCtrl)
+	pvcLister := NewMockPersistentVolumeClaimLister(mockCtrl)
+	pvLister := NewMockPersistentVolumeLister(mockCtrl)
+
+	resourceManager, err := manager.NewResourceManager(podLister, secretLister, configMapLister, serviceLister, pvcLister, pvLister)
+	if err != nil {
+		t.Fatal("Unable to prepare the mocks for resourceManager", err)
+	}
+
+	fakeVolumeSecret := "fake-volume-secret"
+
+	secretLister.EXPECT().Secrets(podNamespace).Return(secretNamespaceLister)
+	secretNamespaceLister.EXPECT().Get(fakeVolumeSecret).Return(&v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fakeVolumeSecret,
+			Namespace: podNamespace,
+		},
+		Data: map[string][]byte{
+			"azurestorageaccountname": []byte("azure storage account name"),
+			"azurestorageaccountkey":  []byte("azure storage account key"),
+		},
+	}, nil)
+
+	pvName := "pv-name"
+
+	pvLister.EXPECT().Get(pvName).Return(&v1.PersistentVolume{
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				CSI: &v1.CSIPersistentVolumeSource{
+					Driver: "file.csi.azure.com",
+					NodeStageSecretRef: &v1.SecretReference{
+						Name:      fakeVolumeSecret,
+						Namespace: podNamespace,
+					},
+				},
+			},
+		},
+	}, nil)
+
+	configMapNamespaceLister := NewMockConfigMapNamespaceLister(mockCtrl)
+	configMapLister.EXPECT().ConfigMaps(podNamespace).Return(configMapNamespaceLister)
+	configMapNamespaceLister.EXPECT().Get("kube-root-ca.crt").Return(&v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "kube-root-ca.crt",
+		},
+		Data: map[string]string{
+			"ca.crt": "fake-ca-data",
+			"foo":    "bar",
+		},
+	}, nil)
+
+	provider, err := createTestProvider(aadServerMocker, aciServerMocker, resourceManager)
+	if err != nil {
+		t.Fatal("Unable to create test provider", err)
+	}
+
+	aciServerMocker.OnCreate = func(subscription, resourceGroup, containerGroup string, cg *aci.ContainerGroup) (int, interface{}) {
+		assert.Check(t, is.Equal(fakeSubscription, subscription), "Subscription doesn't match")
+		assert.Check(t, is.Equal(fakeResourceGroup, resourceGroup), "Resource group doesn't match")
+		assert.Check(t, cg != nil, "Container group is nil")
+		assert.Check(t, is.Equal(podNamespace+"-"+podName, containerGroup), "Container group name is not expected")
+		assert.Check(t, cg.ContainerGroupProperties.Containers != nil, "Containers should not be nil")
+		assert.Check(t, is.Equal(1, len(cg.ContainerGroupProperties.Containers)), "1 Container is expected")
+		assert.Check(t, is.Equal("nginx", cg.ContainerGroupProperties.Containers[0].Name), "Container nginx is expected")
+		assert.Check(t, is.Equal(1, len(cg.Volumes)), "volume count not match")
+		assert.Check(t, is.Equal(pvName, cg.Volumes[0].Name), "volume name doesn't match")
+
```

Comment:
please follow the 

expected, actual style assert/compare. I have written extensive examples in the caravel repo

Reply:
Same, followed the code convention. Happy to use expected/actual instead.